conans/cli/commands/search.py

```diff
@@ -5,38 +5,71 @@
 from conans.cli.command import conan_command, Extender
 
 
-def output_search_cli(info):
-    for remote_info in info:
-        source = "cache" if remote_info["remote"] is None else str(remote_info["remote"])
-        cli_out_write("{}:".format(source), Color.BRIGHT_WHITE)
-        for conan_item in remote_info["items"]:
-            reference = conan_item["recipe"]["id"]
-            cli_out_write(" {}".format(reference))
+def _print_output(info, show_colors):
+    if info["error"]:
+        # TODO: Handle errors
+        return
 
+    if "results" not in info or not info["results"]:
+        return
 
-def output_search_json(info):
+    multiple_remotes = len(info["results"]) > 1
+    indentation = 2 if multiple_remotes else 0
+    remote_color = Color.BRIGHT_BLUE if show_colors else None
+    recipe_color = Color.BRIGHT_WHITE if show_colors else None
+    reference_color = Color.WHITE if show_colors else None
+
+    for result in info["results"]:
+
+        # We only show the name of the remotes if there are multiple remotes
+        if multiple_remotes:
+            cli_out_write("{}:".format(result["remote"]), remote_color)
+
+        if "items" not in result or not result["items"]:
+            cli_out_write("{}There are no matching recipes".format(" " * indentation))
+            continue
+
+        current_recipe = None
+        for recipe in result["items"]:
+            if recipe["recipe"]["name"] != current_recipe:
+                current_recipe = recipe["recipe"]["name"]
+                cli_out_write("{}{}".format(" " * indentation, current_recipe), recipe_color)
+
+            reference = recipe["recipe"]["id"]
+            cli_out_write("{}{}".format(" " * (indentation + 2), reference), reference_color)
+
+
+def _output_search_cli(info):
+    _print_output(info, show_colors=True)
+
+
+def _output_search_raw(info):
+    _print_output(info, show_colors=False)
```

Comment:
Not sure about the ``show_colors`` parametrization. Colorama approach is to add always colors, and if colors are not enabled, then it will filter out the escape caracters. So color is something that is configurable at the app level, maybe not necessary to implement 2 outputs for each command, one with color and one without.

Reply:
I removed the whole `raw` formatter. It doesn't add any value to this command.